### PR TITLE
Wrapped and implemented fsOpenBisFilesystem()

### PR DESCRIPTION
**Pull request by klockee**
Implemented fsOpenBisFilesystem()

### CODE REVIEW

File: nx/source/services/fs.c [hunk 1]
```diff
@@ -101,6 +101,43 @@ Result fsOpenBisStorage(FsStorage* out, u32 PartitionId) {
     return rc;
 }
 
+Result fsOpenBisFileSystem(FsFileSystem* out, u32 PartitionId, const char* string) {
+    IpcCommand c;
+    ipcInitialize(&c);
+    ipcAddSendStatic(&c, string, FS_MAX_PATH, 0);
+    struct {
```
Comment (yellows8) [inline nx/source/services/fs.c]: add newline

**Comment by rajkosto** [comment]
If youre taking in a string pointer, either take also a length or do a strlen(str)+1 in the buffer send, to avoid sending garbage over IPC (this function doesnt really need to take the string pointer though, since official software always just passes 1 zero byte to the IPC)

File: nx/source/services/fs.c [hunk 2]
```diff
@@ -101,6 +101,44 @@ Result fsOpenBisStorage(FsStorage* out, u32 PartitionId) {
     return rc;
 }
 
+Result fsOpenBisFileSystem(FsFileSystem* out, u32 PartitionId, const char* string) {
+    IpcCommand c;
+    ipcInitialize(&c);
```
Comment (yellows8) [inline nx/source/services/fs.c]: Add: char tmpstr[FS_MAX_PATH] = {0};

File: nx/source/services/fs.c [hunk 3]
```diff
@@ -101,6 +101,44 @@ Result fsOpenBisStorage(FsStorage* out, u32 PartitionId) {
...
+    ipcAddSendStatic(&c, string, strlen(string) + 1, 0);
```
Comment (yellows8) [inline nx/source/services/fs.c]: strncpy(tmpstr, string, sizeof(tmpstr)-1);

Then ipcAddSendStatic with tmpstr and sizeof(tmpstr).

File: nx/source/services/fs.c [hunk 4]
```diff
@@ -104,38 +104,41 @@ Result fsOpenBisStorage(FsStorage* out, u32 PartitionId) {
...
+    strncpy(tmpstr, string, sizeof(tmpstr)-1);
+    ipcAddSendStatic(&c, tmpstr, sizeof(tmpstr), 0);
+   
     struct {
         u64 magic;
         u64 cmd_id;
         u32 PartitionId;
     } *raw;
-
+
```
Comment (fincs) [inline nx/source/services/fs.c]: Revert these whitespace changes

**Pull request merged**